lib/lz4.c

```diff
@@ -240,25 +240,70 @@ static const int LZ4_minLength = (MFLIMIT+1);
 /*-************************************
 *  Types
 **************************************/
-#if defined(__cplusplus) || (defined (__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L) /* C99 */)
+#if defined(__VMS)
+# /* nothing */
+#elif defined(_MSC_VER)
+# if (_MSC_VER >= 1600) /* MSVC 2010 */
+#  define LZ4_HAVE_STDINT_H
+# endif
+#elif defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L) /* C99 */
+# define LZ4_HAVE_STDINT_H
+#elif defined(__cplusplus)
+# define LZ4_HAVE_STDINT_H
+#else
+# /* nothing */
+#endif
+
+#include <limits.h>
+
+#if defined(LZ4_HAVE_STDINT_H)
 # include <stdint.h>
+# define LZ4_STDINT_H_INCLUDED
+#else
+# if defined(INT_LEAST32_MAX) && defined(INT8_MAX) && defined(INT32_MAX) && defined(INTPTR_MAX)
+#  define LZ4_STDINT_H_INCLUDED   /* stdint.h already included */
+# endif
+#endif
+
+#if defined(SIZE_MAX)
+# define LZ4_SIZE_MAX  SIZE_MAX
+#else
+# if defined(__WORDSIZE) && (__WORDSIZE == 64)
+#  define LZ4_SIZE_MAX  18446744073709551615ULL
+# endif
+# if defined(__WORDSIZE) && (__WORDSIZE == 32)
+#  define LZ4_SIZE_MAX  4294967295UL
+# endif
+#endif
+
+#ifdef LZ4_STDINT_H_INCLUDED
   typedef  uint8_t BYTE;
   typedef uint16_t U16;
   typedef uint32_t U32;
   typedef  int32_t S32;
   typedef uint64_t U64;
   typedef uintptr_t uptrval;
+  typedef  intptr_t sptrval;
```

Comment:
`sptrval` is declared, but not used.
In which places would it be useful to use it ?

Reply:
A small part of such places:
https://github.com/lz4/lz4/blob/e55548b0bd553cc25d8c0994db6f5d49663fb3f5/lib/lz4hc.c#L144
https://github.com/lz4/lz4/blob/e55548b0bd553cc25d8c0994db6f5d49663fb3f5/lib/lz4hc.c#L317
https://github.com/lz4/lz4/blob/e55548b0bd553cc25d8c0994db6f5d49663fb3f5/lib/lz4hc.c#L615
https://github.com/lz4/lz4/blob/e55548b0bd553cc25d8c0994db6f5d49663fb3f5/lib/lz4hc.c#L647